docs/guides/implementation_blueprint/data-synchronization.md

```diff
@@ -0,0 +1,200 @@
+
+Read our [customer import guide](https://support.voucherify.io/article/67-how-to-import-my-customers#csv-import "How to import my customers?") to learn more.
+
+<!-- Should I write a full step-by-step manual for this? -->
+
+#### Creating and updating customers via the API
+
+If you want to keep your customers database up to date with Voucherify, create a customer in Voucherify every time a new user is added to your database. Use the [Create customer](ref:create-customer) API endpoint to create customer data.
+
+To update customer data, use [Update customer](ref:update-customer) API endpoint. The [Create customer](ref:create-customer) API endpoint can be also used to update customer data.
+
+<!-- Simply repeat the endpoints? -->
+
+#### Synchronizing with connectors
+
+Customer data can be upserted with integrated platforms that support outbound traffic:
+- Batch
+- Bloomreach Engagement
+- Intercom
+- Klaviyo
+- MoEngage
+- mParticle
+- Salesforce
+- Segment
+- ZOHO
+
+<!-- Again, clarification: do we want to add only those platforms that send customer 
+data to Voucherify, right? 
+
+I'll update the links once these get clarified
+
+Also: how to know which platform sends data to V% and which receives V% data?
+
+-->
+
+### Customer API endpoints
+
+<!-- TBChecked which are still supported and if this list is still up-to-date; taken from https://success.voucherify.io/article/430-crm-playbook -->
+
+The following endpoints can be used to synchronize customer data:
+
+- [List Customers](ref:list-customers)
+- [Get Customer](ref:get-customer)
+- [List Customer Activities](ref:list-customer-activities)
+- [List Customer's Segments](ref:list-customer-segments)
+- [Create Customer](ref:create-customer)
+- [Import and Update Customers using CSV](ref:import-customers-using-csv)
+- [Update Customer](ref:update-customer)
+- [Update Customer's Consents](ref:update-customers-consents)
+- [Update Customer’s consents (client-sde)](ref:update-customers-consents-client-side)
+- [Update Customers in bulk](ref:update-customers-in-bulk)
+- [Update Customers' Metadata in bulk](ref:update-customers-metadata-in-bulk)
+- [Delete Customer](ref:delete-customer)
+- [Delete Customer Permanently](ref:delete-customer-permanently)
+
+For reference, see also [the customer object](ref:customer-object) and [customer activity object](ref:customer-activity-object).
+
+<!-- do we also list all other endpoints that have a customer object that can upsert data, e.g. redeem stackable discount, check eligibility? -->
+
+## Product synchronization
+
+Products can be stored in Voucherify but it is not required. The product validation <!-- validation? --> takes place during the following actions:
+- Qualification
+- Validation
+- Redemption
+
+The product validation is based on the API payload and the data stored in Voucherify's product catalog. However, if a product does not exist in Voucherify, the validation will take into consideration the payload alone.
```

Comment:
We can expand here a bit.

While it's possible to synchronize the products, it might be exhausting and requiring lot's of API requests to keep the inventory up to date. Instead of updating the products frequently, you can rely on sending the product data in the qualification, validation, redemption and create order requests only. The verification of the eligibility will consider the data you shared in the request without the need to keep it updated.

Reply:
I've rephrased the whole Section. I'm not sure if this is `good enough` or requires some further polishing.

> Products can be stored in Voucherify but it is not required. The product validation takes place during the following actions:
> - Create order
> - Qualification
> - Redemption
> - Validation
> 
> The product validation is based on the API payload and the data stored in Voucherify's product inventory.
> 
> However, if a product does not exist in Voucherify, the validation will use the product data included in the payload. Even if the product does exist in Voucherify, the payload data are used instead of those stored in the system. This method can be used to avoid frequent API calls to keep the product inventory up to date.
> 
> > 🚧 Product data in the request
> >
> > Unlike customer data, product data provided in the request payload alone do not create a new product in Voucherify. If you want to store products in Voucherify, they need to be created manually in the dashboard or imported with a CSV file.
> > 
> > Also, even though Voucherify uses the payload data instead of the data stored in the system, the product data is not upserted through requests. Products can be updated through product edit function or by importing a CSV file.
> >
> > To enable product updates through the payload, use the [override attribute](#overriding-product-data).
> 
> The products in Voucherify can be grouped into static or dynamic collections based on their parameters. Collections can be used in validation rules to model product-specific limits and loyalty-earning rules.